src/ra_server_sup_sup.erl

```diff
@@ -104,32 +104,59 @@ delete_server(NodeId) ->
     end.
 
 delete_server_rpc(RaName) ->
-    ?INFO("Deleting server ~w and it's data directory.~n",
+    ?INFO("Deleting server ~w and its data directory.~n",
           [RaName]),
     %% TODO: better handle and report errors
     UId = ra_directory:uid_of(RaName),
     Pid = ra_directory:where_is(RaName),
     ra_log_meta:delete(UId),
     Dir = ra_env:server_data_dir(UId),
     _ = supervisor:terminate_child(?MODULE, UId),
-    % TODO: move into separate retrying process
-    try ra_lib:recursive_delete(Dir) of
-        ok ->
-            _ = ra_directory:unregister_name(UId),
-            %% forcefully clean up ETS tables
-            catch ets:delete(ra_log_metrics, UId),
-            catch ets:delete(ra_log_snapshot_state, UId),
-            catch ets:delete(ra_metrics, RaName),
-            catch ets:delete(ra_state, RaName),
-            catch ets:delete(ra_open_file_metrics, Pid),
-            ok
-    catch
-        _:_ = Err ->
-            ?WARN("delete_server/1 failed to delete directory ~s~n"
-                  "Error: ~p~n", [Dir, Err])
-    end,
+    delete_data_directory(Dir),
+    _ = ra_directory:unregister_name(UId),
+    %% forcefully clean up ETS tables
+    catch ets:delete(ra_log_metrics, UId),
+    catch ets:delete(ra_log_snapshot_state, UId),
+    catch ets:delete(ra_metrics, RaName),
+    catch ets:delete(ra_state, RaName),
+    catch ets:delete(ra_open_file_metrics, Pid),
     ok.
 
+delete_data_directory(Directory) ->
+    DeleteFunction = fun() ->
+                         try ra_lib:recursive_delete(Directory) of
+                            ok ->
+                                % moving on
+                                ok
+                         catch
+                            _:_ = Err ->
+                                ?WARN("delete_server/1 failed to delete directory ~s~n"
+                                    "Error: ~p~n", [Directory, Err]),
+                                error
+                         end
+                     end,
+    case DeleteFunction() of
+        ok ->
+            ok;
+        _ ->
+            spawn(fun() ->
+                      retry(DeleteFunction, 2)
+                  end)
+    end.
+
+retry(_Func, 0) ->
+    exhausted;
+retry(Func, Attempt) ->
+    % do not retry immediately
+    timer:sleep(100),
```

Comment:
I'd set a larger default sleep here I think, say 5s. Also maybe move the retry function into `ra_util`